### PR TITLE
added single article page

### PR DESCRIPTION
**Pull request by vincentriley**

### CODE REVIEW

File: src/components/SingleArticlePage.jsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    const {article_id} = useParams()
+    const [article, setArticle] = useState({})
+    const navigate = useNavigate()
+
+    useEffect(() => {
+        fetchData({endpoint: `/articles/${article_id}`})
+        .then(({article}) => {
+            setArticle(article)
+        })
+    })
```
Comment (Poonam-raj) [inline src/components/SingleArticlePage.jsx]: we're now entering error territory where someone could navigate in the page path to '/articles/999999' without using the articles list page at all. Make sure you're throwing some kind of simple 404 message at least when this happens.

Currently all this happens when i navigate to a incorrect article ID:
image

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
A side note: would love to see loading patterns implemented in your app (in articles and single article) at this point and going fwd, make sure you add them in.

Please address the error handling side of things and re-send a PR when you're done

File: src/components/SingleArticlePage.jsx (hunk 1 above)
Comment (vincentriley) [reply src/components/SingleArticlePage.jsx]: should i not be covering this in ticket 12?

Comment (Poonam-raj) [reply src/components/SingleArticlePage.jsx]: You're right it is in ticket 12 but thanks for doing it now, I still think we should address it now

Comment (vincentriley) [reply src/components/SingleArticlePage.jsx]: yeah makes sense :)

**Review by Poonam-raj — APPROVED** [review]
All good, thanks for adding error handling for invalid article ID

**Pull request merged**